src/main/java/com/imaee/propinq/neighborhood/services/implementations/LocalityService.java

```diff
@@ -0,0 +1,109 @@
+        if (localityRequestProvince.isEmpty()) {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, IProvinceRepository.MSG_NOT_EXISTS);
+        }
+
+        Optional<Locality> existingLocality = localityRepository.findByNameIgnoreCaseAndProvince(
+                localityRequest.name(),
+                localityRequestProvince.get()
+        );
+        if (existingLocality.isPresent()) {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, ILocalityRepository.MSG_ALREADY_EXISTS);
+        }
+
+        Locality locality = LocalityMapper.toLocality(
+                localityRequest.name(),
+                localityRequestProvince.get()
+        );
+
+        localityRepository.save(locality);
+
+        return ResponseEntity.ok().build();
+    }
+
+    @Override
+    public ResponseEntity<List<LocalityResponse>> getLocalities() {
+        return ResponseEntity.ok(
+                localityRepository
+                        .findAll()
+                        .stream()
+                        .map(LocalityMapper::toLocalityResponse)
+                        .collect(Collectors.toList())
+        );
+    }
+
+    @Override
+    public ResponseEntity<LocalityResponse> getLocality(UUID id) {
+        Optional<Locality> existingLocality = localityRepository.findById(id);
+        if (existingLocality.isEmpty()) {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, ILocalityRepository.MSG_NOT_EXISTS);
+        }
+
+        return ResponseEntity.ok(LocalityMapper.toLocalityResponse(existingLocality.get()));
+    }
+
+    @Override
+    public  ResponseEntity<Void> updateLocality(UUID id, LocalityRequest localityRequest) {
+        Optional<Province> localityRequestProvince = provinceRepository.findById(localityRequest.provinceId());
+        if (localityRequestProvince.isEmpty()) {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, IProvinceRepository.MSG_NOT_EXISTS);
+        }
+
+        Optional<Locality> existingLocality = localityRepository.findById(id);
+        if (existingLocality.isEmpty()) {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, ILocalityRepository.MSG_NOT_EXISTS);
+        }
+
+        Locality locality = existingLocality.get();
+
+        locality.setName(localityRequest.name());
+        locality.setProvince(localityRequestProvince.get());
+
+        localityRepository.save(locality);
+
+        return ResponseEntity.ok().build();
+    }
+
+    @Override
+    public ResponseEntity<Void> deleteLocality(UUID id) {
+        Optional<Locality> existingLocality = localityRepository.findById(id);
+        if (existingLocality.isEmpty()) {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, ILocalityRepository.MSG_NOT_EXISTS);
+        }
+
+        localityRepository.deleteById(id);
+
+        return ResponseEntity.ok().build();
+    }
```

Comment:
Acá lo mismo que veníamos diciendo

```suggestion
    @Override
    public void deleteLocality(UUID id) {
        throwExceptionIfLocalityNotExistsById
        localityRepository.deleteById(id);
    }
    
    private void throwExceptionIfLocalityNotExistsById(LocalityRequest localityRequest) {
        if (!localityRepository.existsById(localityRequest.getId())
            throw new ResponseStatusException(BAD_REQUEST, MSG_EXISTS);
    }
```